src/biogeochem/Fan2CTSMMod.F90

```diff
@@ -0,0 +1,1059 @@
+    type(soilstate_type)                   , intent(in)    :: soilstate_inst
+    type(temperature_type)                 , intent(in) :: temperature_inst
+    type(waterfluxbulk_type)               , intent(in)    :: waterfluxbulk_inst
+    type(frictionvel_type)                 , intent(in) :: frictionvel_inst
+
+    ! Local variables
+    integer, parameter :: &
+         ! Use this many sub-steps. This improves numerical accuracy but is perhaps not
+         ! essential, because FAN includes an ad-hoc fixer for negative fluxes.
+         num_substeps = 4, &
+         ! FAN includes a separate nitrogen conservation check, which can be done every
+         ! nth time step. This is mostly redundant, because the FAN pools and fluxes are
+         ! now included in the main CLM soil N balance check. The FAN check has more
+         ! detail.
+         balance_check_freq = 1000
+    ! Number of organic N types (available, unavailable, resistant)
+    integer, parameter :: num_org_n_types = 3
+    integer :: c, g, patchcounter, p, status, c1, c2, l, fc, ind_substep
+    real(r8) :: dt, watertend, ratm, tandep
+    
+    ! Organic (non-urea) manure N: production, pools, flux to TAN. Named indices to denote
+    ! each N fraction.
+    real(r8) :: ndep_org(num_org_n_types), orgpools(num_org_n_types), tanprod(num_org_n_types)
+    ! Temporary arrays for N fluxes and pools. Named indices to denote different fluxes;
+    ! numerical indices to denote the age class.
+    real(r8) :: fluxes(num_fluxes, num_cls_max), tanpools(num_cls_max), fluxes_tmp(num_fluxes)
+    ! timestep-cumulative (gN/m2) aging flux out of the oldest class
+    real(r8) :: n_residual, n_residual_total
+    ! H ion concentrations for grz and slr pools (== prescribed + soil pH for oldest class)
+    real(r8) :: Hconc_grz(num_cls_grz), Hconc_slr(num_cls_slr)
+    real(r8) :: fert_inc_tan, pH_soil, pH_crop
+
+    logical :: do_balance_checks
+    real(r8) :: tg, theta, thetasat, infiltr_m_s, evap_m_s, runoff_m_s, org_n_tot, &
+         nstored_old, nsoilman_old, nsoilfert_old, fert_to_air, fert_to_soil, fert_total, fert_urea, fert_tan, &
+         soilflux_org, urea_resid
+    ! TAN production flux from urea. Include one extra flux for the residual flux out of U2.
+    real(r8) :: tanprod_from_urea(num_cls_urea + 1)
+    real(r8) :: ureapools(num_cls_urea)
+    real(r8) :: fract_urea, fract_no3, soilph_min, soilph_max, &
+         soilpsi, fert_no3, fert_generic, bsw
+    integer :: def_ph_count
+    
+    Hconc_grz(1:num_cls_grz-1) = Hconc_grz_def
+    Hconc_slr(1:num_cls_slr-1) = Hconc_slr_def
+
+    soilph_min = 999
+    soilph_max = -999
+    def_ph_count = 0
+    dt = real(get_step_size(), r8)
+    do_balance_checks = balance_check_freq > 0 .and. mod(get_nstep(), balance_check_freq) == 0
+
+    associate(&
+         ngrz => soilbiogeochem_nitrogenflux_inst%manure_n_grz_col, &
+         man_u_grz => soilbiogeochem_nitrogenstate_inst%manure_u_grz_col, &
```

Comment:
We generally try to avoid shortening names in associate, to aid searchability. So, for example, `man_u_grz` should be named `manure_u_grz` (it's okay to leave off the final `_col`).